.github/workflows/_get-screenshots.yaml

```diff
@@ -0,0 +1,100 @@
+name: Grab screenshots
+
+on:
+  workflow_call:
+    inputs:
+      channel:
+        description: "The channel to publish to"
+        default: "candidate"
+        required: true
+        type: string
+      environment:
+        description: "The environment to execute the action in"
+        default: "Candidate Branch"
+        required: true
+        type: string
+      issue-number:
+        description: "The number of the call for testing issue"
+        required: true
+        type: string
+    secrets:
+      SNAPCRAFTERS_BOT_COMMIT:
+        required: true
+
+env:
+  SNAP_NAME: ${{ github.event.repository.name }}
+
+jobs:
+  get-screenshots:
+    name: 📸 Gather screenshots
+    environment: ${{ inputs.environment }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v4
+
+      - name: Setup LXD
+        uses: canonical/setup-lxd@v0.1.1
+
+      - name: Setup vmctl
+        run: |
+          wget -qO vmctl https://raw.githubusercontent.com/snapcrafters/ci/main/tools/vmctl
+          chmod 755 ./vmctl
+
+      - name: Prepare VM
+        run: |
+          ./vmctl prepare
+          ./vmctl exec "sudo snap install ${SNAP_NAME} --channel latest/${{ inputs.channel }} --classic"
+          ./vmctl exec "snap run ${SNAP_NAME} &>/home/ubuntu/${SNAP_NAME}.log &"
+          sleep 60
+
+      - name: Gather screenshots
+        run: |
+          ./vmctl screenshot-full
+          ./vmctl screenshot-window
+
+      - name: Output application logs
+        run: |
+          ./vmctl exec "cat /home/ubuntu/${SNAP_NAME}.log"
+
+      - name: Checkout snapcrafters/ci-screenshots
+        uses: actions/checkout@v4
+        with: 
+          repository: snapcrafters/ci-screenshots
+          path: ci-screenshots
+          token: ${{ secrets.SNAPCRAFTERS_BOT_COMMIT }}
+
+      - name: Upload screenshots to snapcrafters/ci-screenshots
```

Comment:
Yeh, we had a discussion about this in Telegram. External hosting services require secrets generally, and unfortunately artifacts get uploaded as a zip file, which means they cannot be hot linked in the comment.

What's missing right now is the automation to periodically clean out that repo by date - I think we'll set it up so that anything older than 90 days get's auto-pruned or similar.

Reply:
Additionally, GitHub actions artifacts are a paid feature. Repo's are not. So it's better for us to use the repo, even if it's not the most "clean" solution.